File: Retos/Reto #3 - EL GENERADOR DE CONTRASEÑAS [Media]/java/Qv1ko.java
import java.util.ArrayList;

public class Qv1ko {

    public static void main(String[] args) {
        passwordGenerator(21,true,false,false);
    }//main

    private static void passwordGenerator(int length,boolean uppercase,boolean numbers,boolean symbols) {
        String password="";
        length=(length<8)? 8:(length>16)? 16:length;
        ArrayList<Character> characters=new ArrayList<Character>();
        for(int i=97;i<=122;i++) {
            characters.add((char)i);
        }
        if(uppercase) {
            for(int i=65;i<=90;i++) {characters.add((char)i);}
        }
        if(numbers) {
            for(int i=48;i<=57;i++) {characters.add((char)i);}
        }
        if(symbols) {
            for(int i=33;i<=47;i++) {characters.add((char)i);}
            for(int i=58;i<=64;i++) {characters.add((char)i);}
            for(int i=91;i<=96;i++) {characters.add((char)i);}
            for(int i=123;i<=126;i++) {characters.add((char)i);}
        }
        for(int i=0;i<length;i++) {
            password+=characters.get((int)(Math.random()*characters.size()));
        }
        System.out.println(password);
    }//passwordGenerator

}//class

File: Retos/Reto #3 - EL GENERADOR DE CONTRASEÑAS [Media]/typescript/Qv1ko.ts
passGen(10, true, false, true)

function passGen(len: number, upper: boolean, num: boolean, simb: boolean) {

    let password = ""

    let chars = ["a", "b", "c", "d", "e", "f", "g", "h", "i", "j", "k", "l", "m", "n", "o", "p", "q", "r", "s", "t", "u", "v", "w", "x", "y", "z"]
    const upperLetters = ["A", "B", "C", "D", "E", "F", "G", "H", "I", "J", "K", "L", "M", "N", "O", "P", "Q", "R", "S", "T", "U", "V", "W", "X", "Y", "Z"]
    const numbers = ["0", "1", "2", "3", "4", "5", "6", "7", "8", "9"]
    const simbols = ["`", "~", "!", "@", "#", "$", "%", "^", "&", "*", "(", ")", "-", "_", "=", "+", "[", "{", "]", "}", "\\", "|", ";", ":", "'", "\"", ",", "<", ".", ">", "/", "?"]

    len = (len < 8)? 8 : (len > 16)? 16 : len;

    if (upper) {
        chars.push(...upperLetters)
    }

    if (num) {
        chars.push(...numbers)
    }

    if (simb) {
        chars.push(...simbols)
    }

    for (let i = 0; i < len; i++) {
        password += chars[Math.floor(Math.random() * chars.length)]
    }

    console.log(password)

}
